### ISSUE TITLE
Unable to serialize a List property in response entity 

### ISSUE THREAD

**Issue by hantsy** [OpenLiberty/open-liberty#15845]
I created a simple EJB branch in my [jakartaee8-starter](https://github.com/hantsy/jakartaee8-starter-boilerplate/tree/ejb) project which is provided as a template for developers to kickstart a Jakarta EE 8 project.

The `PurchaseOrder` contains a list of `OrderItem`.

The `/orders` endpoint (OrderResource) will return all orders, but when hitting it, it throws exceptions:

```
[INFO] [ERROR   ] Unable to serialize property 'items' from com.example.cart.PurchaseOrder
[INFO] [ERROR   ] Generating incomplete JSON
[INFO] [ERROR   ] Problem with writing the data, class java.util.Vector, ContentType: application/json
[INFO] [err] java.lang.Throwable: ANDY
[INFO] [err]    at org.apache.cxf.jaxrs.utils.JAXRSUtils.logMessageHandlerProblem(JAXRSUtils.java:2055)
[INFO] [err]    at [internal classes]
[INFO] [err]    at javax.servlet.http.HttpServlet.service(HttpServlet.java:686)
[INFO] [err]    at com.ibm.websphere.jaxrs.server.IBMRestServlet.service(IBMRestServlet.java:96)
[INFO] [err]    at [internal classes]
[INFO] [err]    at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[INFO] [err]    at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[INFO] [err]    at java.base/java.lang.Thread.run(Thread.java:834)
[INFO] [WARNING ] Interceptor for {http://cart.example.com/}OrderResource has thrown exception, unwinding now
[INFO] Unable to serialize property 'items' from com.example.cart.PurchaseOrder
[INFO] [WARNING ] Could not create a secure Stax XMLInputFactory.  Found class com.sun.xml.internal.stream.XMLInputFactoryImpl.  Suggest Woodstox 4.2.0 or newer.
```

Attached my message log here.
[messages.log](https://github.com/OpenLiberty/open-liberty/files/5949186/messages.log)

**Comment by andymc12** [comment]
Hi @hantsy - first off, sorry about the strange throwable there. I use "ANDY" as an eye-catcher when I'm debugging, but that never should have been checked in.  I have already checked in a change to remove it.

The problem here is that the JSON-B detected a circular/recursive reference in the entities that it is serializing - towards the bottom of the log file you will see:
```
Caused by: javax.json.bind.JsonbException: Unable to serialize property 'items' from com.example.cart.PurchaseOrder
	at org.eclipse.yasson.internal.serializer.ObjectSerializer.serializeInternal(ObjectSerializer.java:71)
        ...
Caused by: javax.json.bind.JsonbException: Unable to serialize property 'order' from com.example.cart.OrderItem
	at org.eclipse.yasson.internal.serializer.ObjectSerializer.serializeInternal(ObjectSerializer.java:71)
        ...
Caused by: javax.json.bind.JsonbException: Recursive reference has been found in class class com.example.cart.PurchaseOrder.
	at org.eclipse.yasson.internal.serializer.ObjectSerializer.serializeInternal(ObjectSerializer.java:76)
        ...
``` 

So `PurchaseOrder` has a field, `items` that references `OrderItem`.  And `OrderItem` has a field, `order` that references `PurchaseOrder`.  JSON-B doesn't know how to handle this.  According to https://stackoverflow.com/questions/61915478/how-solve-bidirectional-relationships-on-jsonb-serialization , the way around this is to add `@JsonbTransient` to one of those fields.

Hope this helps!

**Comment by hantsy** [comment]
@andymc12 I realized this issue, use a DtoAssembler to convert it to dtos(no cycle deps). Thanks.

**Comment by srl295** [comment]
hi ANDY and thanks! :tada:
just for the record, looks like the fix went into 7cb975ec2d74da0209be7b0e70336808e5d5a7a3
